app/routes/reroute-guid.js

```diff
@@ -0,0 +1,10 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
+    init() {
```

Comment:
Use `beforeModel` instead of `init`. Init will only fire once on page load. `beforeModel` will work in either case. Also, with `beforeModel`, you get the `transition` parameter, instead of trying to parse the pathname.
```js
    beforeModel(transition) {
        const {state, targetName} = transition;
        const slug = state.params[targetName].bad_url.replace(/\/.*$/, '');

        if (slug.length === 5)
            window.location.href = `${window.location.origin}/${slug}`;
    }
```